dotnet/src/SemanticKernel/Text/JsonSerializerContexts.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Planning;
+using Microsoft.SemanticKernel.SemanticFunctions;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Text;
+
+[JsonSerializable(typeof(Plan))]
+[JsonSerializable(typeof(PromptTemplateConfig))]
+[JsonSerializable(typeof(SKFunction))]
+internal sealed partial class CoreSourceGenerationContext : JsonSerializerContext
+{
+    public static readonly CoreSourceGenerationContext WithGeneralOptions = new(new JsonSerializerOptions(Json.GeneralOptions));
+    public static readonly CoreSourceGenerationContext Indented = new(new JsonSerializerOptions { WriteIndented = true });
+}
+
+[JsonSourceGenerationOptions(IncludeFields = true)]
+[JsonSerializable(typeof(Plan))]
+internal sealed partial class IncludeFieldsSourceGenerationContext : JsonSerializerContext
```

Comment:
My understanding from @eiriktsarpalis is that IncludeFields is special in that it needs to be visible to the source generator as it impacts the generated code. Eirik?

Reply:
This comment isn't critical to be addressed in this PR. But GH isn't giving me a way to mark it as "resolved". Unless @eiriktsarpalis suggests something that should be changed, this can be "resolved".